requirements-cuda.txt

```diff
@@ -2,6 +2,7 @@
 -r requirements-common.txt
 
 # Dependencies for NVIDIA GPUs
+cupy-cuda12x
```

Comment:
Might be good to have a separate requirements.txt for aDAG-only requirements.

Reply:
+1. Maybe create requirements-adag.txt and add ray >=2.32 and this? we should also check a runtime check